test_defopt.py

```diff
@@ -598,9 +598,14 @@ def test_implicit_noparser(self):
         def notok(foo):
             """:type foo: NotConstructibleFromStr"""
 
-        with self.assertRaises(Exception):
+        with self.assertRaises(Exception) as context:
             defopt.run(notok, argv=["foo"])
 
+        self.assertEqual(
+            str(context.exception),
+            "no parser found for type NotConstructibleFromStr",
+        )
```

Comment:
I would write this check as using assertRaisesRegex.

Reply:
Updated and softened the pattern a bit.